File: app/src/main/java/com/kkkkkn/readbooks/view/activities/StackManager.java
package com.kkkkkn.readbooks.view.activities;

import android.app.Activity;

import java.util.ArrayList;
import java.util.List;

public class StackManager {
    private static volatile StackManager stackManager;
    private final List<Activity> activityList=new ArrayList<>();

    public static StackManager getInstance(){
        if(stackManager==null){
            synchronized (StackManager.class){
                if(stackManager==null){
                    stackManager=new StackManager();
                }
            }
        }
        return stackManager;
    }

    public void addActivity(Activity activity){
        activityList.add(activity);
    }

    public void exitAllActivity(){
        for (int i = 0; i <activityList.size() ; i++) {
            activityList.get(i).finish();

        }
        activityList.clear();
    }

    public void removeActivity(Activity activity){
        activityList.remove(activity);
    }
}

File: app/src/main/java/com/kkkkkn/readbooks/view/activities/StackManager.kt
package com.kkkkkn.readbooks.view.activities

import android.app.Activity

class StackManager {
    private val activityList: MutableList<Activity> = ArrayList()
    fun addActivity(activity: Activity) {
        activityList.add(activity)
    }

    fun exitAllActivity() {
        for (i in activityList.indices) {
            activityList[i].finish()
        }
        activityList.clear()
    }

    fun removeActivity(activity: Activity) {
        activityList.remove(activity)
    }

    companion object {
        @Volatile
        private var stackManager: StackManager? = null
        val instance: StackManager?
            get() {
                if (stackManager == null) {
                    synchronized(StackManager::class.java) {
                        if (stackManager == null) {
                            stackManager =
                                StackManager()
                        }
                    }
                }
                return stackManager
            }
    }
}